api-registration/server/index.js

```diff
@@ -0,0 +1,66 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  const sql = `
+  insert into "users" ("username", "hashedPassword")
+    values ($1, $2)
+    returning "userId", "username", "createdAt"
+  `;
+  argon2
+    .hash(password)
+    .then(hashedPassword => {
+      const params = [username, hashedPassword]
+      db.query(sql, params)
+      .then(result => {
+        const output = result.rows
```

Comment:
The output should just be the user object, this will output an array with the user object in it. You can use array destructuring or just `[0]`.

```JAVASCRIPT
const [ output ] = result.rows;

// --- OR ---

const output = result.rows[0];
```